tools/devops/automation/templates/build/build.yml

```diff
@@ -1,230 +1,252 @@
+        env:
+          CREDENTIALS: $(dotnetclimsrc-read-sas-token)
+        displayName: "Configure install extra args"
+        timeoutInMinutes: 5
+
+  - bash: $(System.DefaultWorkingDirectory)/xamarin-macios/tools/devops/automation/scripts/bash/configure-build.sh
     env:
-      Token: $(dn-bot-dnceng-artifact-feeds-rw)
+      ${{ if eq(parameters.isPR, true) }}:
+        IsPR: 'True'
+      ${{ else }}:
+        IsPR: 'False'
+    displayName: "Configure build"
+    workingDirectory: "$(Build.SourcesDirectory)/xamarin-macios"
+    timeoutInMinutes: 5
 
   - bash: |
       set -x
       set -e
-      ARGS="--azure-feed \"https://dotnetclimsrc.blob.core.windows.net/dotnet\" -FeedCredential $CREDENTIALS"
-      echo "##vso[task.setvariable variable=DOTNET_INSTALL_EXTRA_ARGS]$ARGS"
+      cat configure.inc
+    displayName: "Print configuration"
+    workingDirectory: "$(Build.SourcesDirectory)/xamarin-macios"
+    timeoutInMinutes: 5
+
+  # Make sure we have the right maccore hash checked out before we try to add
+  # the provisioning profiles.
+  - bash: |
+      set -ex
+      time make -C $(Build.SourcesDirectory)/xamarin-macios/ reset
+    name: resetDependencies
+    displayName: 'Reset dependencies'
+    timeoutInMinutes: 10
+
+  # We'll need these profiles to build the hot restart prebuilt app during the build
+  # (it's built for device, and thus needs a certificate available so that the app can be signed).
+  - bash: ./install-qa-provisioning-profiles.sh -v
+    displayName: 'Add build provisioning profiles'
+    timeoutInMinutes: 30
+    continueOnError: true  # should not stop the build
+    workingDirectory: $(Build.SourcesDirectory)/maccore/tools
     env:
-      CREDENTIALS: $(dotnetclimsrc-read-sas-token)
-    displayName: "Configure install extra args"
+      AUTH_TOKEN_GITHUB_COM: ${{ parameters.gitHubToken }}
+      AUTH_TOKEN_LA_DEV_APPLE_P12: ${{ parameters.xqaCertPass }}
+      AUTH_TOKEN_LA_DISTR_APPLE_P12: ${{ parameters.xqaCertPass }}
+      AUTH_TOKEN_LA_MAC_INSTALLER_DISTR_P12: ${{ parameters.xqaCertPass }}
+      AUTH_TOKEN_VSENG_XAMARIN_MAC_DEVICES_P12: ${{ parameters.xqaCertPass }}
+      AUTH_TOKEN_VSENG_XAMARIN_MAC_DEVICES_2_P12: ${{ parameters.xqaCertPass }}
+
+  - bash: |
+       set -x
+       set -e
+       rm -rf ~/Library/Developer/Xcode/DerivedData/*
+       rm -rf ~/Library/Caches/com.apple.dt.Xcode
+    displayName: 'Clear Xcode cache'
+
+  # only install all the simulators when working on a PR, EO machines do not allow use to perform this step
+  - ${{ if parameters.isPR }}:
+      - bash: $(Build.SourcesDirectory)/xamarin-macios/system-dependencies.sh --provision-simulators --ignore-shellcheck --ignore-yamllint
+        displayName: 'Provision simulators (system-dependencies.sh)'
+  - ${{ else }}:
+      - bash: |
+          curl -o iOS_18_beta_3_Simulator_Runtime.dmg https://dalexsotofiles.blob.core.windows.net/xcode/iOS_18_beta_3_Simulator_Runtime.dmg
```

Comment:
Download the runtime from a known url.

Reply:
How is this 'known url' protected? If we can, we should probably get this behind an authenticated url.